tools/autograd/derivatives.yaml

```diff
@@ -1183,6 +1183,9 @@
 - name: upsample_nearest3d(Tensor self, int[3] output_size, float? scales_d=None, float? scales_h=None, float? scales_w=None) -> Tensor
   self: upsample_nearest3d_backward(grad, output_size, self.sizes(), scales_d, scales_h, scales_w)
 
+- name: upsample_nearest2d.vec(Tensor input, int[]? output_size, float[]? scale_factors) -> Tensor
```

Comment:
Since "int[]?" does not have the length of output_size anymore either,  the signature of all versions of upsample (1d-2d-3d) are now the same.
The plan is to simplify this further to have a single function for all upsample cases?
If so would that be done in this PR as well? if not what is the reason not to?

Reply:
Very good point.  One possible downside would be increased code size on mobile because we currently do selective builds at operator granularity, and bundling them together would mean that someone who just wants to use upsample_nearest2d would also get the code for 1d and 3d.  We have a plan to mitigate this, but I'm not sure when we'll have time to implement it.  I'll measure the code size impact and report back.